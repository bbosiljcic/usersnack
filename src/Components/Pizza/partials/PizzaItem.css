.pizza-item {
    margin-bottom: 2rem;
    display: flex;
}

.pizza-item--content {
    flex-grow: 1;
    padding: 2rem;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.pizza-item--incredients {
    color: #757575;
    margin-bottom: 2rem;
}

.pizza-item--price {
    font-size: 2rem;
    font-weight: 700;
}

.pizza-item--actions {
    display: flex;
    justify-content: space-between;
}