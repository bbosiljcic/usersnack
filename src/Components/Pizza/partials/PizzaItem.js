import React from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import { Button, Card } from '@material-ui/core';

import './PizzaItem.css';
import joinArrayPhonetically from '../../../helpers/formating';

export default function PizzaItem(props) {
  const { pizza, isDetail } = props;
  const history = useHistory();

  return (
    <Card className="pizza-item">
      <div className="pizza-item--content">
        <div>
          <h3>{pizza.name}</h3>
          <span className="pizza-item--incredients">{joinArrayPhonetically(pizza.incredients)}</span>
        </div>
        <div className="pizza-item--actions">
          {
            !isDetail
            && (
            <Button variant="outlined" color="primary" onClick={() => history.push(`/pizza/${pizza.id}`)}>
              Order a {pizza.name}
            </Button>
            )
          }
          <div className="pizza-item--price">
            USD {pizza.price.toFixed(2)}
          </div>
        </div>
      </div>
      <img src={`/assets/pizzas/${pizza.img}`} alt={`tasty ${pizza.name}`} />

    </Card>
  )
}

PizzaItem.propTypes = {
  pizza: PropTypes.shape({
    id: PropTypes.number,
    img: PropTypes.string,
    incredients: PropTypes.array,
    name: PropTypes.string,
    price: PropTypes.number,
  }).isRequired,
  isDetail: PropTypes.bool,
};
