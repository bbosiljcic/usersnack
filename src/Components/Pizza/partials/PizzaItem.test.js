import React from 'react';
import renderer from 'react-test-renderer';

import PizzaItem from './PizzaItem';

it('renders correctly when there a pizza', () => {
  const pizza = {
    id: 1,
    img: test.jpg,
    incredients: ['ing1', 'ing2', 'ing3'],
    name: 'test Pizza',
    price: 100,
  }
  const tree = renderer.create(<PizzaItem pizza={pizza} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there a pizza and isDetail', () => {
  const pizza = {
    id: 1,
    img: test.jpg,
    incredients: ['ing1', 'ing2', 'ing3'],
    name: 'test Pizza',
    price: 100,
  }
  const tree = renderer.create(<PizzaItem pizza={pizza} isDetail />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there a pizza and is detail', () => {
  const pizza = {
    id: 1,
    img: test.jpg,
    incredients: ['ing1', 'ing2', 'ing3'],
    name: 'test Pizza',
    price: 100,
  }
  const tree = renderer.create(<PizzaItem pizza={pizza} isDetail />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there a pizza, is detail and has a price', () => {
  const pizza = {
    id: 1,
    img: test.jpg,
    incredients: ['ing1', 'ing2', 'ing3'],
    name: 'test Pizza',
    price: 100,
  }
  const tree = renderer.create(<PizzaItem pizza={pizza} isDetail price={200} />).toJSON();
  expect(tree).toMatchSnapshot();
});
