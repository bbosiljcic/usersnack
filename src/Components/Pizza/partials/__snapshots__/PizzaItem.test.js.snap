// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there a pizza 1`] = `
<div
  className="MuiPaper-root MuiCard-root pizza-item MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="pizza-item--content"
  >
    <div>
      <h3>
        test Pizza
      </h3>
      <span
        className="pizza-item--incredients"
      >
        ing1, ing2 and ing3
      </span>
    </div>
    <div
      className="pizza-item--actions"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Order a 
          test Pizza
        </span>
      </button>
      <div
        className="pizza-item--price"
      >
        $ 
        100.00
      </div>
    </div>
  </div>
  <img
    alt="tasty test Pizza"
    src="/assets/pizzas/undefined"
  />
</div>
`;

exports[`renders correctly when there a pizza and is detail 1`] = `
<div
  className="MuiPaper-root MuiCard-root pizza-item MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="pizza-item--content"
  >
    <div>
      <h3>
        test Pizza
      </h3>
      <span
        className="pizza-item--incredients"
      >
        ing1, ing2 and ing3
      </span>
    </div>
    <div
      className="pizza-item--actions"
    >
      <div
        className="pizza-item--price"
      >
        $ 
        100.00
      </div>
    </div>
  </div>
  <img
    alt="tasty test Pizza"
    src="/assets/pizzas/undefined"
  />
</div>
`;

exports[`renders correctly when there a pizza and isDetail 1`] = `
<div
  className="MuiPaper-root MuiCard-root pizza-item MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="pizza-item--content"
  >
    <div>
      <h3>
        test Pizza
      </h3>
      <span
        className="pizza-item--incredients"
      >
        ing1, ing2 and ing3
      </span>
    </div>
    <div
      className="pizza-item--actions"
    >
      <div
        className="pizza-item--price"
      >
        $ 
        100.00
      </div>
    </div>
  </div>
  <img
    alt="tasty test Pizza"
    src="/assets/pizzas/undefined"
  />
</div>
`;

exports[`renders correctly when there a pizza, is detail and has a price 1`] = `
<div
  className="MuiPaper-root MuiCard-root pizza-item MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="pizza-item--content"
  >
    <div>
      <h3>
        test Pizza
      </h3>
      <span
        className="pizza-item--incredients"
      >
        ing1, ing2 and ing3
      </span>
    </div>
    <div
      className="pizza-item--actions"
    >
      <div
        className="pizza-item--price"
      >
        $ 
        200.00
      </div>
    </div>
  </div>
  <img
    alt="tasty test Pizza"
    src="/assets/pizzas/undefined"
  />
</div>
`;
